cpp/src/gandiva/precompiled/string_ops.cc

```diff
@@ -261,6 +263,44 @@ char* castVARCHAR_utf8_int64(int64 context, const char* data, int32 data_len,
   return ret;
 }
 
+#define CAST_VARCHAR(IN_TYPE, FORMAT_TYPE)                                           \
+  FORCE_INLINE                                                                       \
+  const char* castVARCHAR##_##IN_TYPE(int64 context, IN_TYPE num, int32* out_len) {  \
+    *out_len = snprintf(NULL, 0, "%" FORMAT_TYPE, num);                              \
```

Comment:
Several things:
* `snprintf` is locale-dependent; this is probably not important for integers (but will be if you add support for reals)
* no need to call `snprintf` twice, IMHO; just use a large enough local array (e.g. 50 bytes will certainly be enough)
* we have utilties in `arrow/util/formatting.h` which avoid the use of `snprintf`

Reply:
Thanks, I missed the utility functions in arrow that does that. I will refactor the code to use those.